### DISCUSSION TITLE
shap TreeExplainer

### DISCUSSION THREAD

**Discussion by RainerHilscher** [sebp/scikit-survival#266] [General]
Hi scikit-survival team,

do you have any plans to make GradientBoostingSurvivalAnalysis or RandomSurvivalForest compatible with the shap TreeExplainer?

Currently (0.17.1) I get the following exception: Model type not yet supported by TreeExplainer: <class 'method'>

Rainer

**Comment by sebp** [comment]
I haven't tried myself, but according to #213 it should work.

**Reply by RainerHilscher** [reply]
#213 was very instructive ... including your .predict trick. However, it only covers the shap Explainer class, NOT the TreeExplainer.
